### PR TITLE
Feature/cc 2825

### PR DESCRIPTION
**Pull request by rDr4g0n**
Fixing https://jira.zenoss.com/browse/CC-2825

When stop, start, or restart are clicked for a service, a request is made to the backend to count the number of descendants that will be affected by the change. However, if the network connection between the UI and serviced is slow, then the user can end up waiting for a number of seconds. The user needs immediate feedback that something is happening so that they know their click was received. 

This change does a few things to give the user immediate feedback on the click:

1) Shows a "one moment please" modal while waiting on the child count
2) Shows a modal even if the number of affected services is only 1  

modal
(note that this gif shows the best case scenario where the descendant count loads within 200ms. In some cases that loading dialog may be visible for many seconds)

To support these features, a few changes needed to be made

* A new "unclosable" modal feature was added so that the "one moment please" modal cant just be closed, resulting in unexpected behavior
* When modals are displayed one-after-another, the UI flashes in a yucko way, so the bootstrap modal background was removed and replaced with our own modal background, which lets us control how it shows/hides more closely
* some codemirror textarea issues popped up in modals with this change, so those were tightened up
* most of this code needs to be shared with the Apps page, so it was moved into a shared location

### CODE REVIEW

File: web/ui/src/Services/ServiceDetailsController.js [hunk 1]
```diff
@@ -329,81 +329,35 @@
                 // children and ask the user to choose to start all
                 // children or only the top service
                 $scope.clickRunning = function (service, state) {
```
Comment (rDr4g0n) [inline web/ui/src/Services/ServiceDetailsController.js]: this all gets moved into `Service.js` as a class method

File: web/ui/src/Services/ServiceDetailsController.js [hunk 2]
```diff
@@ -329,81 +329,35 @@
...
+                    let onStartService = function(modal){
+                        // the arg here explicitly prevents child services
+                        // from being started
+                        $scope.setServiceState(service, state, true);
+                        modal.close();
+                    };
+                    let onStartServiceAndChildren = function(modal){
+                        $scope.setServiceState(service, state);
+                        modal.close();
+                    };
```
Comment (rDr4g0n) [inline web/ui/src/Services/ServiceDetailsController.js]: This is all moved into `modalService.js` as a shared modal

File: web/ui/src/Services/ServiceDetailsController.js [hunk 3]
```diff
@@ -495,6 +449,7 @@
                                 ],
                                 onShow: function () {
                                     modalScope.codemirrorRefresh = true;
+                                    modalScope.$apply();
```
Comment (rDr4g0n) [inline web/ui/src/Services/ServiceDetailsController.js]: it takes a tick or 2 for this view to update, so update it asap

**Comment by tgjonestx** [comment]
Retest this

**Comment by tgjonestx** [comment]
retest this

**Comment by rDr4g0n** [comment]
retest this

**Pull request merged**